internal/ca/helper.go

```diff
@@ -0,0 +1,197 @@
+		Name:    cm.Name,
+	}
+
+	return &caCM, nil
+}
+
+func (h *helperImpl) GetServiceCA(ctx context.Context, namespace string) (*ConfigMap, error) {
+	cm, err := h.getConfigMap(ctx, namespace, serviceCAType)
+	if err != nil {
+		return nil, fmt.Errorf("could not get CA ConfigMap: %v", err)
+	}
+
+	caCM := ConfigMap{
+		KeyName: "service-ca.crt",
+		Name:    cm.Name,
+	}
+
+	return &caCM, nil
+}
+
+func (h *helperImpl) getOrCreateClusterCAConfigMap(ctx context.Context, namespace string) (*v1.ConfigMap, error) {
+	cm, err := h.getConfigMap(ctx, namespace, clusterCAType)
+	if err != nil {
+		if !errors.Is(err, errConfigMapNotFound) {
+			return nil, fmt.Errorf("could not get the cluster CA ConfigMap: %v", err)
+		}
+
+		cm = &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "kmm-cluster-ca-",
+				Namespace:    namespace,
+				Labels: map[string]string{
+					typeKey: clusterCAType,
+					"config.openshift.io/inject-trusted-cabundle": "true",
+				},
+			},
+		}
+
+		if err = h.client.Create(ctx, cm); err != nil {
+			return nil, fmt.Errorf("could not create the cluster CA ConfigMap: %v", err)
+		}
+	}
+
+	return cm, nil
+}
+
+func (h *helperImpl) getOrCreateServiceCAConfigMap(ctx context.Context, namespace string) (*v1.ConfigMap, error) {
+	cm, err := h.getConfigMap(ctx, namespace, serviceCAType)
+	if err != nil {
+		if !errors.Is(err, errConfigMapNotFound) {
+			return nil, fmt.Errorf("could not get the cluster CA ConfigMap: %v", err)
+		}
+
+		cm = &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "kmm-service-ca-",
+				Namespace:    namespace,
+				Annotations:  map[string]string{"service.beta.openshift.io/inject-cabundle": "true"},
+				Labels:       map[string]string{typeKey: serviceCAType},
+			},
+		}
+
+		if err = h.client.Create(ctx, cm); err != nil {
+			return nil, fmt.Errorf("could not create the cluster CA ConfigMap: %v", err)
+		}
+	}
+
+	return cm, nil
+}
+
+func (h *helperImpl) getConfigMap(ctx context.Context, namespace, caType string) (*v1.ConfigMap, error) {
+	cmList := v1.ConfigMapList{}
+
+	opts := []client.ListOption{
+		client.InNamespace(namespace),
+		client.MatchingLabels{typeKey: caType},
+	}
+
+	if err := h.client.List(ctx, &cmList, opts...); err != nil {
+		return nil, fmt.Errorf("could not list ConfigMaps: %v", err)
+	}
+
+	if len(cmList.Items) == 0 {
+		return nil, errConfigMapNotFound
+	}
+
+	if l := len(cmList.Items); l > 1 {
+		return nil, fmt.Errorf("expected 1 ConfigMap, got %d", l)
+	}
+
+	return &cmList.Items[0], nil
+}
+
+func (h *helperImpl) syncConfigMapOwners(ctx context.Context, cm *v1.ConfigMap, owner client.Object, logger logr.Logger) error {
+	logger = logger.WithValues("cm-name", cm.Name)
+
+	logger.Info("Reconciling CA ConfigMap owners")
+
+	p := client.MergeFrom(cm.DeepCopy())
+
+	if err := controllerutil.SetOwnerReference(owner, cm, h.scheme); err != nil {
+		return fmt.Errorf("could not set %s as the owner of the ConfigMap: %v", owner, err)
```

Comment:
Probably same as [here](https://github.com/rh-ecosystem-edge/kernel-module-management/pull/361/files#r1069533833) for the `owner` ?